WIP: docs(TextField): Add TextField component documenation

wichniowski opened the pull request:
TextField component documentation.

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

## Doc Checklist

- [x] Provide all configurations
- [ ] Provide copy guidelines

wichniowski commented:
@m10l I'd do the the docs of all components which are part of TextField and then document the TextField as a composition of those components wdyt?

m10l commented:
> @m10l I'd do the the docs of all components which are part of TextField and then document the TextField as a composition of those components wdyt?

Sounds good!

wichniowski commented:
@m10l yeah I'm meeting later today with Sara

wichniowski commented:
@m10l @joaoramos I added docs for HelpText, ValidationMessage, FormLabel, TextInput and TextField as the composition out of those components. I'll add TextArea as well. Other component docs like CopyButton which is also used in the TextField component will happen in a sperate PR. @sarakalinoski is reviewing copy and opened up her own branch. So lets get the review started :). Some open questions:

- Should we add a section for a11y consideration per component, right now I've added it to the best practices.
- How to call "compositional properties" like textInputProps.. maybe compositional properties is ok.
- For all interactive elements, we need to provide some kind of a note that event handlers are provided, so we don't have to mention obvious props like onClick every time for button or links. I think this should stay part of the storybook wdyt.

sarakalinoski commented:
> * Should we add a section for a11y consideration per component, right now I've added it to the best practices.

Can we see how many A11y considerations we have for each component and decide from there? If it's just one it makes sense to wrap it under another section, but if we have more, why not make it it's own section.

wichniowski commented:
@sarakalinoski yeah I think this is the best way to deal with it.

sarakalinoski commented on packages/forma-36-website/src/pages/components/text-field/index.mdx:
```diff
@@ -0,0 +1,172 @@
+
+## Best practices
+
+- Provide a labelText, name and id to ensure accessibility
```
Don't need "a" and start with objective:

To ensure TextField meets accessibility standards, provide labelText, name and id.

sarakalinoski commented on packages/forma-36-website/src/pages/components/text-field/index.mdx:
```diff
@@ -0,0 +1,172 @@
+
+## TextField variations
+
+The TextField component can be configured in a number of different ways. Here is a guide to when to use certain variations.
```
The TextField component can be configured in a number of different ways. Use this guidance to determine which variation to use based on the context.

wichniowski commented:
@sarakalinoski wow thanks a lot for your review and all those great commits :)
